File: leetcode/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        """
        SLIDING WINDOW
        RUNTIME: 83 ms (81.16%)
        MEMORY: 13 MB (49.67%)
        """

        max_ = 0
        i = len(s)

        while i > 0:
            for k in range(len(s)-i+1):
                current = s[k:k+i]
                if len(current) == len(set(current)):
                    max_ = max(max_, i)
                    break
            if i < max_:
                break
            i -= 1

        return max_

sol = Solution()
sol.lengthOfLongestSubstring('abcabcbb')

File: leetcode/0003-longest-substring-without-repeating-characters/solution.cpp
class Solution {
public:
    /**
     * RUNTIME: 11 ms (90.89%)
     * MEMORY: 8.3 MB (61.28%)
    */
    int lengthOfLongestSubstring(string s) {
        unordered_map <char, int> mp;
        int ans = 0;
        int i = 0;
        for (int j=0; j < s.size(); j++) {
            if (mp.find(s[j]) != mp.end()) {
                i = max(i, mp[s[j]]);
            }
            ans = max(ans, j - i + 1);
            mp[s[j]] = j + 1;
        }
        return ans;
    }
};
